<a href="https://www.viharatech.com"><img src="https://miro.medium.com/v2/resize:fit:828/format:webp/1*on2Bpj6A5gfd-CzaFOAZRA.jpeg"></a>


- `Welcome to our interactive learning session! In this Jupyter notebook, you'll find a series of questions related to Python List Concept. To maximize your learning, please write Python code in the cell below each question and execute it. This hands-on practice will help reinforce your understanding of the concepts. Feel free to reach out if you have any questions or need assistance. Happy coding! 💻🐍 `

### Question 1: Write a Python program to count the number of elements in a list without using len.


- `sample input : `
    - `var = [10,11,12,13]`
- `sample output :`
    - `result : 4`

In [ ]:
# Please write The solution Here for above question 1:
var = []
count=0
n = int(input("Enter number of elements: "))
for i in range(n):
    element = input(f"Enter element {i+1}: ")
    var.append(element)
for i in var:
  count+=1
print(count)

Enter number of elements: 4
Enter element 1: 10
Enter element 2: 11
Enter element 3: 12
Enter element 4: 13
4


### Question 2: Write a Python program to find the maximum and minimum values in a given list.

- `sample input : `
    - `var = [10,11,12,13,14,15]`
- `sample output :`
    - `result : max = 15 , min = 10`

In [ ]:
# Please write The solution Here for above question 2:
var = []
n = int(input("Enter number of elements: "))
for i in range(n):
    element = input(f"Enter element {i+1}: ")
    var.append(element)
sorted_var = sorted(var)
min = sorted_var[0]
max = sorted_var[-1]
print(f"min is {min} and max is {max}")

Enter number of elements: 3
Enter element 1: 134
Enter element 2: 121
Enter element 3: 155
min is 121 and max is 155


### Question 3: Write a Python program to remove the first occurrence of a specified element from a list.

- `sample input : `
    - `var = [1,2,3,4,2,5]`
- `sample output :`
    - `result : [1,3,4,2,5]`
    

In [ ]:
# Please write The solution Here for above question 3:
var = []
n = int(input("Enter number of elements: "))
for i in range(n):
    element = int(input(f"Enter element {i+1}: "))
    var.append(element)
x = int(input("enter the element you want to remove : "))
var.remove(x)
print(var)

Enter number of elements: 6
Enter element 1: 1
Enter element 2: 2
Enter element 3: 3
Enter element 4: 4
Enter element 5: 2
Enter element 6: 5
enter the element you want to remove : 2
[1, 3, 4, 2, 5]


### Question 4: Write a Python program to find the average of all the elements in a list.


- `sample input : `
    - `var = [1,2,3,4,2,5]`
- `sample output :`
    - `result : 2.833 `

In [ ]:
# Please write The solution Here for above question 4:
var = []
sum = 0
n = int(input("Enter number of elements: "))
for i in range(n):
    element = int(input(f"Enter element {i+1}: "))
    var.append(element)
for i in var:
  sum += i
average = round(sum/n, 3)
print(f"Average of numbers in the list is : {average}")

Enter number of elements: 6
Enter element 1: 1
Enter element 2: 2
Enter element 3: 3
Enter element 4: 4
Enter element 5: 5
Enter element 6: 2
Average of numbers in the list is : 2.833


### Question 5: Write a Python program to check if a list is empty or not.


- `sample input : `
    - `var1 = [1,2]`
- `sample output :`
    - `result : Not empty`
    
- `sample input : `
    - `var1 = []`
- `sample output :`
    - `result :empty`
    


In [ ]:
# Please write The solution Here for above question 5:
var = []
n = int(input("Enter number of elements: "))
for i in range(n):
    element = int(input(f"Enter element {i+1}: "))
    var.append(element)
if var:
  print(f"not empty, list is : {var}")
else:
  print("empty")


Enter number of elements: 2
Enter element 1: 1
Enter element 2: 2
not empty, list is : [1, 2]


### Question 6:  Write a Python program to find the index of a specified element in a list..

- `sample input : `
    - `var = [1,2,3,4,2,5]`
- `sample output :`
    - `result : 3 index is 2`

In [ ]:
# Please write The solution Here for above question 6:
var = []
n = int(input("Enter number of elements: "))
for i in range(n):
    element = int(input(f"Enter element {i+1}: "))
    var.append(element)
x = int(input("enter the element to find it's index : "))
y = var.index(x)
print(f" {x} index is  {y}")


Enter number of elements: 6
Enter element 1: 1
Enter element 2: 2
Enter element 3: 3
Enter element 4: 4
Enter element 5: 2
Enter element 6: 5
enter the element to find it's index3
 3 index is  2


### Question 7: Write a Python program to find which number is palindrome in below given list .

- `sample input : `
    - `var = [134,121,155]`
- `sample output :`
    - `result : 134 is not a palindome , 121 is a palindrome `

In [ ]:
# Please write The solution Here for above question 7:
var1 = []
var2 = []
n = int(input("Enter number of elements: "))
for i in range(n):
    element = int(input(f"Enter element {i+1}: "))
    var1.append(element)
for number in var1:
  string_number =  str(number)
  if string_number == string_number[::-1]:
    var2.append(number)
print(f"here is the list of palimdromes : {var2}")


here is the list of palimdromes : [121, 34543, 878, 91019]


### Question 8: Write a Python program to find the common elements between two lists.

- `sample input : `
    - `var1 = [1,2,3,4,10]`
    - `var2 = [10,11,12,14]`
- `sample output :`
    - `result : 10`

In [ ]:
# Please write The solution Here for above question 8:
var1 = [1,2,3,4,10]
var2 = [10,11,12,14]
var3 = []
for number1 in var1:
  for number2 in var2:
    if number1 == number2:
      var3.append(number1)
print(f"common elements are : {var3}")



common elements are : [10, 11, 14, 12]


### Question 9: Write a Python program to remove the duplicates from a list

- `sample input : `
    - `var = [10,11,12,10,12,24]`
- `sample output :`
    - `result : [10,11,12,24]`

In [ ]:
# Please write The solution Here for above question 9:
var1 = [10,11,12,10,12,24]
var2 = []
for number in var:
  if number not in var2:
    var2.append(number)
print(var2)

[10, 11, 12, 24]


### Question 10:  Write a Python program to sort a list of strings in Ascii Order [Ascending order]


- `sample input : `
    - `var = ['a' , 'c' , 'b']`
- `sample output :`
    - `result : ['a' , 'b' , 'c']`


In [37]:
# Please write The solution Here for above question 10:
var1 = ['a' , 'c' , 'b','e', 'd']
var2 = []
var3 = []
for i in  var1:
  var2.append(ord(i))
var2 = sorted(var2)
for i in var2:
  var3.append(chr(i))
print(var3)


['a', 'b', 'c', 'd', 'e']
